Homework 11 - Chapter 11 - Nguyen Chinh Quan - 220066

In [15]:
import pandas as pd
import math

11.14.

Table: 
- k - 1 = 3
- SSA = 4.522 * (k-1) = 4.522 * 3 = 13.566
- SSE = 62 * 0.988 = 61.256
- SST = 13.566 + 61.256 = 74.822
- F = MSA / MSE = = 4.522 / 0.988 = 4.577
- p  value = 0.00585 (using online calculator)

a. 
- H0: mean1 = mean2 = mean3 = mean4 (all population means are equal)
- Ha: meani != meanj for some i != j

b.

- F(0.01,3,62) = 4.11419
- Rejection region for a = 0.01: F >= 4.11419

c.

4.577 > 4.11419, therefore, there is evidence to suggest that at least two population means are different.

11.15.

In [16]:
data_15 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 11\ex11-15chips.xls")
data_15

BBQ  Cheese  Olestra  Baked
0  338     235      164    290
1  155     238      197    343
2  239     251      136    294
3  184     229      214    373
4  185     233      148    306
5  261     232      230    357

n = 24, k = 4

a.
- H0: meanBBQ = meanCheese = meanOlestra = meanBaked
- Ha: meani != meanj for some i != j

b.


In [17]:
sum_15 = data_15.sum().sum()
SST_15 = (data_15**2).sum().sum() - (sum_15**2) / 24 
SSA_15 = ((((data_15.sum())**2)/6).sum()) - (sum_15**2) / 24 
SSE_15 = SST_15 - SSA_15
MSA_15 = SSA_15 / (4-1)
MSE_15 = SSE_15 / (24 - 4)
F_15 = MSA_15 / MSE_15
print(SST_15, SSA_15, SSE_15, MSA_15, MSE_15, F_15)

103040.0 67000.33333333349 36039.66666666651 22333.444444444496 1801.9833333333256 12.393812990008005


Using online calculator, we have p value = 0.000084

c. 
- alpha = 0.05
- F(0.05,3,20) = 3.0984
- 12.393812990008005 > 3.0984, therefore, there is evidence to suggest that the population mean amount of salt per serving is different for at least two varieties. 

11.17.

- H0: meanChicago = meanHonolulu = meanNewYork = meanSeattle = meanWashington
- Ha: meani != meanj for some i != j

In [18]:
sample_size_17 = 14+18+20+15+19
sum_17 = 463 + 620 + 506 + 356 + 586
SST_17 = (16185 + 22458 + 13686 + 8964 + 18840) - (sum_17**2)/sample_size_17
SSA_17 = (463**2/14 + 620**2/18 + 506**2/20 + 356**2/15 + 586**2/19)- (sum_17**2)/sample_size_17
SSE_17 = SST_17 - SSA_17
MSA_17 = SSA_17 / 4
MSE_17 = SSE_17 / (sample_size_17 - 5)
F_17 = MSA_17 / MSE_17
print(F_17)


7.35492551099679


- alpha = 0.05
- F(0.05,4,81) = 2.48444
- 7.35492551099679 > 2.48444, therefore, there is evidence that the population mean times through some pair of traffic bottlenecks are different.

11.28.

In [19]:
data_28 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 11\ex11-28expense.xls")
data_28

Shanghai  Singapore  Bejing  Hong Kong  Mexico City  Montreal
0      23.74      22.36   28.54      25.06        18.69     26.45
1       6.96      14.27   24.25      41.96        17.20     29.48
2      30.50      24.77   11.52      17.77        27.55     30.21
3      14.64      16.46   12.54      19.47        16.44     23.64
4       9.13      20.82   20.68      18.13        35.36     26.79
5      12.35      13.76   22.95      35.05        27.01     18.62
6      16.86      31.80   16.08      19.84        22.84     30.83
7      17.44      26.20   20.94      37.86        15.08     29.78
8      16.13      21.19   21.72      31.76         2.86     26.22
9      12.51      21.53   19.57      13.68        27.30     30.24
10     23.34      14.30   20.99      11.71        17.58     31.19
11     13.34      14.29   13.61      20.32        25.84     16.43
12     18.32      14.30   28.60      25.91        23.22     35.48
13      6.30      16.71   23.19      21.03        20.98     28.45
14     10.74      17.91   17.81      16.12          NaN     26.78
15       NaN      15.91   20.73      29.71          NaN       NaN
16       NaN        NaN   11.25      29.99          NaN       NaN
17       NaN        NaN   13.36      31.23          NaN       NaN
18       NaN        NaN   27.62      21.28          NaN       NaN
19       NaN        NaN   18.69      24.97          NaN       NaN

- H0: meanShanghai = meanSingapore = meanBejing = meanHongkong = meanMexicoCity = meanMontreal
- Ha: meani != meanj for some i != j

In [20]:
sample_size_28 = data_28.count().sum()
sum_28 = data_28.sum().sum()
SST_28 = (data_28**2).sum().sum() - (sum_28**2) / sample_size_28
SSA_28 =  ((data_28["Shanghai"].sum())**2/15 + (data_28["Singapore"].sum())**2/16 + 
           (data_28["Bejing"].sum())**2/20 + (data_28["Hong Kong"].sum())**2/20 + 
           (data_28["Mexico City"].sum())**2/14 + (data_28["Montreal"].sum())**2/15) - (sum_28**2) / sample_size_28
SSE_28 = SST_28 - SSA_28
MSA_28 = SSA_28 / (6-1)
MSE_28 = SSE_28 / (sample_size_28 - 6)
F_28 = MSA_28 / MSE_28
print(F_28)

6.665792546172569


- alpha = 0.01
- F(0.01, 5, 94) = 3.21835
- 6.665792546172569 > 3.21835, therefore, there is evidence to suggest that at least two of the population mean amounts spent on ground transportation are different.

11.44. (Sorry I don't know how to graph)

a. 
- mean1 and mean2 are not significantly different.
- mean1 and mean3 are not significantly different.
- mean2 and mean3 are not significantly different.
- All due to 0 in CI (for each group).

b.
- mean1 and mean2 are significantly different (0 not in CI).
- mean1 and mean3 are not significantly different (0 in CI).
- mean2 and mean3 are not significantly different (0 in CI).

c.
- mean1 and mean2 are significantly different (0 not in CI).
- mean1 and mean3 are not significantly different (0 in CI).
- mean1 and mean4 are not significantly different (0 in CI).
- mean2 and mean3 are not significantly different (0 in CI).
- mean2 and mean4 are significantly different (0 not in CI).
- mean3 and mean4 are not significantly different (0 in CI).

d. 
- mean1 and mean2 are significantly different (0 not in CI).
- mean1 and mean3 are significantly different (0 not in CI).
- mean1 and mean4 are not significantly different (0 in CI).
- mean2 and mean3 are not significantly different (0 in CI).
- mean2 and mean4 are significantly different (0 not in CI).
- mean3 and mean4 are significantly different (0 not in CI).

11.55.

In [21]:
data_55 = pd.read_excel(r"D:\stat\Tex-_Datasets\Chapter 11\ex11-55honey.xls")
data_55

North  South  Midwest  West
0   68.2   70.3     69.3  79.8
1   65.8   76.8     64.4  69.1
2   71.2   78.8     66.9  82.5
3   73.4   68.6     70.0  80.0
4   62.1   72.6     62.5  76.3
5   59.5   77.3     60.2  79.0

- H0: meanNorth = meanSouth = meanMidwest = meanWest
- Ha: meani != meanj  for some i != j

a.

In [22]:
sample_size_55 = data_55.count().sum()
sum_55 = data_55.sum().sum()
SST_55 = (data_55**2).sum().sum() - (sum_55**2) / sample_size_55
SSA_55 = ((((data_55.sum())**2)/6).sum()) - (sum_55**2) / sample_size_55
SSE_55 = SST_55 - SSA_55
MSA_55 = SSA_55 / (4-1)
MSE_55 = SSE_55 / (sample_size_55 - 4)
F_55 = MSA_55 / MSE_55
print(SST_55, SSA_55, SSE_55, MSA_55, MSE_55, F_55)

1034.6449999999604 621.6483333333163 412.99666666664416 207.21611111110542 20.649833333332207 10.034759494965256


- alpha = 0.05
- F(0.05,3,20) = 3.0984
- 10.034759494965256 > 3.0984, therefore, there is evidence of an overall difference in population mean honey production.

b.

Number of pairwise comparisons: 4*3/2 = 6

In [23]:
mean_North_55 = data_55["North"].mean()
mean_South_55 = data_55["South"].mean()
mean_Midwest_55 = data_55["Midwest"].mean()
mean_West_55 = data_55["West"].mean()
print(0.05 / (2*6))

0.004166666666666667


t(0.0042,20) = 2.923557

- North and South:

In [27]:
CI_North_South_55 = ((mean_North_55 - mean_South_55) - 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6), 
                     (mean_North_55 - mean_South_55) + 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6))
CI_North_South_55

(-15.036911267503733, 0.30357793417038526)

meanNorth and meanSouth are not significantly different (0 in CI).

- North and Midwest:

In [28]:
CI_North_Midwest_55 = ((mean_North_55 - mean_Midwest_55) - 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6), 
                     (mean_North_55 - mean_Midwest_55) + 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6))
CI_North_Midwest_55

(-6.520244600837054, 8.820244600837064)

meanNorth and meanMidwest are not significantly different (0 in CI).

- North and West:

In [29]:
CI_North_West_55 = ((mean_North_55 - mean_West_55) - 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6), 
                     (mean_North_55 - mean_West_55) + 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6))
CI_North_West_55

(-18.753577934170387, -3.413088732496269)

meanNorth and meanWest are significantly different (0 not in CI).

- South and Midwest:

In [30]:
CI_South_Midwest_55 = ((mean_South_55 - mean_Midwest_55) - 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6), 
                     (mean_South_55 - mean_Midwest_55) + 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6))
CI_South_Midwest_55

(0.8464220658296204, 16.18691126750374)

meanSouth and meanMidwest are significantly different (0 not in CI).

- South and West:

In [31]:
CI_South_West_55 = ((mean_South_55 - mean_West_55) - 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6), 
                     (mean_South_55 - mean_West_55) + 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6))
CI_South_West_55

(-11.386911267503713, 3.953577934170405)

meanSouth and meanWest are not significantly different (0 in CI).

- Midwest and West:

In [32]:
CI_Midwest_West_55 = ((mean_Midwest_55 - mean_West_55) - 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6), 
                     (mean_Midwest_55 - mean_West_55) + 2.923557*math.sqrt(MSE_55)*math.sqrt(1/6 + 1/6))
CI_Midwest_West_55

(-19.903577934170393, -4.563088732496275)

meanMidwest and meanWest are significantly different (0 not in CI).

c. (Sorry I don't know how to graph)

Not significantly different groups are:
- North and South
- North and Midwest
- South and West


Significantly different groups are:
- North and West
- South and Midwest
- Midwest and West